### PR TITLE
Move UFOGen to research projects

### LINKED ISSUE #10162: Move UFOGen Pipeline and Scheduler to Research Projects
**Is your feature request related to a problem? Please describe.**
After reviewing #6133 and a bit of searching has led me to the implementation of the UFOGen paper by the co-authors. By reading the repo ReadME and it seems there is no model checkpoints provided by the authors, only the training code. Does it make sense to move the pipeline to example/research projects?

Repo link:  https://github.com/xuyanwu/SIDDMs-UFOGen

Please let me know.

cc: @yiyixuxu , @sayakpaul

### PR DESCRIPTION
**Pull request by ParagEkbote**
# What does this PR do?

As discussed previously, I have moved the UFOGen scheduler and the code shared by its co-authors to examples/research projects.

I have also followed the same approach that was talked about in #10046. The implementation added has been kept simplistic to avoid confusion as well. Please let me know if any further corrections are needed and I will make the necessary changes.

Fixes #10162 and closes #6133 

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/diffusers/blob/main/CONTRIBUTING.md)?
- [x] Did you read our [philosophy doc](https://github.com/huggingface/diffusers/blob/main/PHILOSOPHY.md) (important for complex PRs)?
- [x] Was this discussed/approved via a GitHub issue or the [forum](https://discuss.huggingface.co/c/discussion-related-to-httpsgithubcomhuggingfacediffusers/63)? Please add a link to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/diffusers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/diffusers/tree/main/docs#writing-source-documentation).
- [x] Did you write any new necessary tests?

## Who can review?

cc: @yiyixuxu

### CODE REVIEW

**Comment by hlky** [comment]
Not sure research projects is the right place for a clone of an entire repo? Looks like the scheduler is the main thing and that was already added in community scripts.

**Comment by ParagEkbote** [comment]
The issue is that after the release of the UFOGen repo, the authors didn't publicly release a trained model based on the paper and it gets quite difficult to verify the results.  ㋡

The folder contains the code needed to experiment and verify the results in the future, thereby I chose to add the majority of files.

**Comment by ParagEkbote** [comment]
Could you please review?


cc: @yiyixuxu

**Comment by ParagEkbote** [comment]
I need advice on how to proceed further for this PR. Any advice you could give is greatly appreciated.

@yiyixuxu

**Comment by ParagEkbote** [comment]
Could you please review the changes?

cc: @yiyixuxu

**Comment by ParagEkbote** [comment]
Closing this PR for now. If anyone has any feedback or would like to continue working on this, please feel free to reopen the PR or create a new one.
